src/hammer-vlsi/defaults.yml

```diff
@@ -367,3 +367,13 @@ lvs.submit:
   command_meta: lazysubst
   settings: "vlsi.submit.settings"
   settings_meta: lazycrossref
+
+# Key name for the technology stackup
+# This should exist in the stackups list in the tech json
+technology.stackup: null
```

Comment:
They seem generally sane, 3 comments about these:

1) Would be good to add `# type: <type>` comments for each of these to facilitate future documentation scraping
2) Let's put this under `technology.core` maybe? So `vlsi.technology` is about how hammer-vlsi interacts with technology. I'd imagine these settings could go under `technology.<mytech>` since we may want different settings per technology. Take a look at extracted_tarballs_dir?
3) Would put them all under one heading:
```
technology.core:
  # foo bar foo bar
  # type: ...
  stackup: null
  # foo bar
  # type: ...
  std_cell_rail_layer: null
```

Reply:
I think these should go where `placement_site` is, which is currently`vlsi.technology`. I can move `placement_site` if you like `technology.core` better. No preference.

Reply:
So far settings like `placement_site`, this is the order of precedence I have in my head. Say I'm interested in technology property `foobar` (or `placement_site` etc). Then the code will look in the following order (highest to lowest priority):

1. `vlsi.technology.overrides.foobar`
1. `technology.<mytech>.foobar`
2. `technology.core.foobar`, if the above does not exist

This enables a config database of the following e.g.:

In techs.yml
```
# technology defaults
technology.mytech45.foobar: 12345
technology.footech99.foobar: 67876

# any unknown technology, use this
technology.core.foobar: 1000
```

In myproject.yml
```
# in this project use this regardless of technology (for incremental adoption)
technology.core.
vlsi.technology.overrides.foobar: 9001
```

Also, not sure it makes sense for everything in `vlsi.technology` to move to `technology.core` e.g. `extra_macro_sizes` is design-specific.

Reply:
Does this currently exist? I get 2 and 3, but I don't really understand 1. Can't your personal yml just override `technology.<mytech>.foobar`? This seems a little complicated to me.

Reply:
2 and 3 sort of exist (see `extracted_tarballs_dir` for an example). Maybe we don't need (1), but then there's no way to globally override `foobar` which may not really make sense without a technology anyway.

Reply:
I am inclined to leave this as-is then (or at least move things to `vlsi.technology` to be consistent with `placement_site`) if this requires additional pythoning that is not automatic. IMO it's pretty simple to just have separate `myproject.techA.yml` and `myproject.techB.yml` files that separate out technology-specific settings. I actually think it's undesirable to put all possible technology settings in a single file or database for NDA reasons.

Reply:
Filed #317, I think this has uncovered some things we need to take a look at, IMO.

Reply:
I'm going to leave them as they currently are, and we should clean them up with #317 when we get to that.

Reply:
Dealing with this later in #317 is fine.

Would it still be possible for you to add the type annotations and put them under one heading (`technology:` is fine for now)?

Regarding "I actually think it's undesirable to put all possible technology settings in a single file or database for NDA reasons." --> the hammer database is actually aggregated from many separate files, so this "global" override would just be something mixed in. It's not one giant file.

Reply:
...Actually no, I'm going to make them `technology.core.*` because that way if people agree with my thoughts in #317 they need not be changed.

Reply:
For putting them into one YAML heading it doesn't actually matter what it is, no?

```
technology: # or technology.core
  # foo bar foo bar
  # type: ...
  stackup: null
  # foo bar
  # type: ...
  std_cell_rail_layer: null
```

Reply:
oh, yes, I agreed with that change and forgot to mention it. I already fixed that :).

Reply:
Marking this resolved, we can debate in #317 if needed.